hw/arm/xnu.c

```diff
@@ -267,8 +261,15 @@ DTBNode* load_dtb_from_file(char *filename) {
     DTBNode *root = NULL;
     uint8_t *file_data = NULL;
     uint32_t fsize;
+    char payload_type[4];
+
+    extract_im4p_payload(filename, payload_type, &file_data, &fsize);
+
+    if (strncmp(payload_type, "dtre", 4) != 0) {
+        error_report("Could parse ASN.1 data in file '%s' because it is not a 'dtre' object, found '%.4s' object.", filename, payload_type);
```

Comment:
```suggestion
        error_report("Couldn't parse ASN.1 data in file '%s' because it is not a 'dtre' object, found '%.4s' object.", filename, payload_type);
```